app/Jobs/ProcessImage.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Models\Image;
+use Exception;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+use Intervention\Image\Facades\Image as InterventionImage;
+
+class ProcessImage implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(Image $image)
+    {
+        $this->image = $image;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        if ($this->image->path) {
+            Log::debug('Image path already set. Exiting;');
+            return;
+        }
+
+        $destinationPath = "app/images/{$this->image->id}.jpg";
+
+        $processedImage = InterventionImage::make($this->getLocalPath());
+        $processedImage->save(storage_path($destinationPath));
+
+        $this->image->path = $destinationPath;
+        $this->image->save();
+    }
+
+    private function getLocalPath() {
+        if (Str::startsWith($this->image->source_path, '\\\AA20\OddArch\AA20 dáta\OBRÁZKY')) {
```

Comment:
This should actually only go up yo here:
```suggestion
        if (Str::startsWith($this->image->source_path, '\\\AA20\OddArch\AA20 dáta')) {
```

because there are several subdirectries being used here (`OBRÁZKY`, `FOTOGRAFIE`, ...)